scylla/src/transport/connection_pool.rs

```diff
@@ -938,49 +941,94 @@ impl PoolRefiller {
         }
     }
 
+    #[cfg(feature = "cloud")]
+    fn maybe_translate_for_serverless(
+        &self,
+        mut endpoint: UntranslatedEndpoint,
+    ) -> impl Future<Output = UntranslatedEndpoint> {
+        if let Some(cloud_config) = self.pool_config.connection_config.cloud_config.clone() {
```

Comment:
I don't consider `Either` to be any complication. For me, it is an ordinary tool for the job of branched evaluation - both for Futures and iterators. Everyone should be familiar with this usage. After all, this is a major usecase for `Either`.

Reply:
I don't think I agree. Personally, I have never needed to use it and I had to look up what Either means in the context of futures.

I can see some situations where it could be useful, e.g. putting futures of two types into a vector. Here, however, I don't think this is the case. I think you overcomplicated the logic slightly here, using standard language constructs and stdlib should be sufficient to express what you wanted to express.

Suggestion (it should also slightly simplify `start_opening_connection`):

```rust
#[cfg(not(feature = "cloud"))]
async fn maybe_translate_for_serverless(
    _config: &ConnectionConfig,
    endpoint: UntranslatedEndpoint,
) -> impl Future<Output = UntranslatedEndpoint> {
    async move { endpoint }
}

#[cfg(feature = "cloud")]
async fn maybe_translate_for_serverless(
    config: &ConnectionConfig,
    mut endpoint: UntranslatedEndpoint,
) -> impl Future<Output = UntranslatedEndpoint> {
    let cloud_config = config.cloud_config.clone();
    async move {
        if let Some(cloud_config) = cloud_config {
            // ...
        } else {
            // ...
        }
    }
}
```

Reply:
I've been considering before such a solution as you proposed above. What I dislike about it is that it does not regard features as _adding_ something to our code (as in additional `let` that I wrote), but rather _switching_ a whole function to an alternate. If such design is acceptable IYO, then I can adopt it.